File: sheet 9 X. Mountain Scenery/main.py
n, k = map(int, input().split())
arr = list(map(int, input().split()))
for i in range(1, n * 2, 2):
    if arr[i] - arr[i - 1] > 1 and arr[i] - arr[i + 1] > 1:
        arr[i] -= 1
        k -= 1
        if k == 0:
            break
print(*arr)

File: sheet 9 X. Mountain Scenery/main.js
var input = readline().split` `.map(x => +x)
var n = input[0], k = input[1]
var points = readline().split` `.map(x => +x)

while (k) {
    for (var i = 1; i < n * 2 && k; i += 2) {
        if (points[i] > points[i - 1] && points[i] > points[i + 1]) {
            if (points[i] - 1 > points[i - 1] && points[i] - 1 > points[i + 1]) {
                points[i]--;
                k--;
            }
        }
    }
}

print(...points)